### PR TITLE
Allow viewing join link page without login

### LINKED ISSUE #1807: Make Joining Circle Modal appear BEFORE login modal
Currently, when you send someone a “magic community joining link”, they have to sign in before they can see details about the community they can join. Ideally they wouldn’t have to.

Implementing this is non-trivial though:

- You have to be able to fetch data about the community without being authenticated (API changes)
- You have to render the app without being authenticated (component tree changes)
From proposal: https://www.notion.so/coordinape/Show-info-for-community-join-link-before-logging-in-9774703bb4b24a4cb91afe9620b03121)

### PR DESCRIPTION
**Pull request by levity**
Fixes #1807 

### Testing

- Generate a valid join link
  - Load it while logged out; verify the page appears with correct data and a "Connect Wallet" button
  - Click "Connect Wallet" and log in; verify the page re-appears after login, now with a form instead of "Connect Wallet"
  - Submit the form; verify that it adds you to the circle as before
- Load an invalid join link; verify it shows an error page, whether you're logged in or not
- Verify that welcome links still behave correctly (no change)

### TODO

- [x] extract general test environment changes into their own PR

### CODE REVIEW

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1828** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1828" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1828?utm_source=stack-comment).

File: src/utils/testing/index.tsx [hunk 1]
```diff
@@ -24,29 +26,33 @@ export { provider, takeSnapshot, restoreSnapshot };
...
+  return <>{children}</>;
 };
 
 export const TestWrapper = ({
-  children,
+  children = null,
   withWeb3 = false,
+  withRoutes = false,
   queryClientCallback,
+  routeHistory = [],
```
Comment (levity) [inline src/utils/testing/index.tsx:55]:
```suggestion
  routeHistory = ['/'],
```

**Comment by CryptoGraffe** [comment]
@levity can you resolve these conflicts?

**Review by CryptoGraffe — COMMENTED** [review]
This looks fine but I need to test it with SideNav, unless you have already @levity  ?

**Comment by levity** [comment]
> This looks fine but I need to test it with SideNav, unless you have already @levity ?

Seems fine

Screen Shot 2023-01-25 at 8 58 34 PM

There's a text color error in dark mode (the highlighted text is black), but I don't think that's specific to this feature? The "How Email Login Works" modal has the same issue

Screen Shot 2023-01-25 at 8 57 09 PM

Btw, unrelated, but mouseover on the close button for Toasts is messed up

Screen Shot 2023-01-25 at 9 00 50 PM

**Review by CryptoGraffe — APPROVED** [review]

**Pull request merged**